### ISSUE TITLE
array of pointers breaks syntax highlighting

### ISSUE THREAD

**Issue by matter123** [jeff-hykin/cpp-textmate-grammar#128]
Code:
```c++
	void bar();
	entries = new symbolTableEntry *[entriesCount];
	void bar();
```
Image:
<img width="457" alt="Screen Shot 2019-04-26 at 11 28 35 AM" src="https://user-images.githubusercontent.com/714007/56828639-98954780-6816-11e9-8ae7-4c5fbd3de45c.png">

Is fixed by adding `{}` at the end of line 2, so it is likely the fault of the lambda pattern.

**Comment by matter123** [comment]
It looks like * is valid before a lambda ([godbolt](https://godbolt.org/z/rNk7TD)).

Therefore it seems the correct change should be to have the new expression capture the type as well. That's a bit more work and dereferencing a self-executing lambda is probably far less common than an array of pointers. So the lambda pattern should be modified to prevent this.

I'll modify #81 to mention this.

**Comment by jeff-hykin** [comment]
I agree and this code might also be an issue. It would also be fixed by matching the tape before the brackets 

https://github.com/atom/language-c/issues/325

**Comment by matter123** [comment]
It looks fine for me?
<img width="323" alt="Screen Shot 2019-04-26 at 13 41 09" src="https://user-images.githubusercontent.com/714007/56835409-0185bb00-6829-11e9-9393-6f7dce479cc9.png">

```c++
for(auto [x,y]:vec) {
      int z=0;
      if(x>0) {
            // do something
      }
      else {
            // do sometheing else
      }
}
```

**Comment by jeff-hykin** [comment]
Ok that's good, I wasn't sure how the lambda pattern would handle it

**Comment by matter123** [comment]
Yeah its not tagged as a lambda.

```
punctuation.definition.begin.bracket.square.cpp
meta.bracket.square.access.cpp
meta.parens.cpp
source.cpp
```

**Comment by jeff-hykin** [comment]
Temp fix is merged in 👍

**Comment by matter123** [comment]
Temporarily fixed in v1.8.9.
